### PR TITLE
fix(#1494): git showUntracked value and log

### LINKED ISSUE #1494: git runner logging error during table concat with nil
### Description

When **git** option is enabled and the .gitconfig contains no value for status.showUntrackedFiles, an _invalid value (nil) at index 5 in table for 'concat'_ is encountered. This is worked around by adding an appropriate value to the .gitconfig for status.showUntrackedFiles. 

Appears this could be be resolved by handling the nil prior to loggingthis could be handled with a check for the _nil_ value in that table pri

### Neovim version

```text
NVIM v0.8.0-dev-699-gcbfae548e
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```

### Operating system and version

Windows 10

### nvim-tree version

261a5c3

### Minimal config

```lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvt-min/site]]
local package_root = "/tmp/nvt-min/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup {
    {
      "wbthomason/packer.nvim",
      "kyazdani42/nvim-tree.lua",
      "kyazdani42/nvim-web-devicons",
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  }
end
if vim.fn.isdirectory(install_path) == 0 then
  print "Installing nvim-tree and dependencies."
  vim.fn.system { "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path }
end
load_plugins()
require("packer").sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua setup()]]
vim.opt.termguicolors = true
vim.opt.cursorline = true

-- MODIFY NVIM-TREE SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
_G.setup = function()
  require("nvim-tree").setup {
      git = {
        enable = true,
      }
  }
end
```

### Steps to reproduce

1. **.gitconfig** for cwd should have no entry for **status.showUntrackedFiles**
2. _nvim -nu /tmp/nvt-min.lua_
3. _:NvimTreeOpen_
4. Create new file within NvimTree

### Expected behavior

Expect NvimTree to open without error

### Actual behavior

**Encounter error:**

Error executing vim.schedule lua callback: ...\packer\start\nvim-tree.lua/lua/nvim-tree/git/runner.lua:89: invalid value (nil) at index 5 in table for 'concat'
stack traceback:
	[C]: in function 'concat'
	...\packer\start\nvim-tree.lua/lua/nvim-tree/git/runner.lua:89: in function '_run_git_job'
	...\packer\start\nvim-tree.lua/lua/nvim-tree/git/runner.lua:146: in function 'run'
	...ck\packer\start\nvim-tree.lua/lua/nvim-tree/git/init.lua:38: in function 'reload_project'
	...ker\start\nvim-tree.lua/lua/nvim-tree/explorer/watch.lua:10: in function 'reload_and_get_git_project'
	...ker\start\nvim-tree.lua/lua/nvim-tree/explorer/watch.lua:50: in function 'refresh_path'
	...ker\start\nvim-tree.lua/lua/nvim-tree/explorer/watch.lua:68: in function 'callback'
	...\pack\packer\start\nvim-tree.lua/lua/nvim-tree/utils.lua:358: in function <...\pack\packer\start\nvim-tree.lua/lua/nvim-tree/utils.lua:357>

### PR DESCRIPTION
**Pull request by AxLauri**
* should_show_untracked evaluates non-boolean status.showUntrackedFiles

* git.Runner status logging nil values removed before concat

closes #1494

### CODE REVIEW

File: lua/nvim-tree/git/runner.lua [hunk 1]
```diff
@@ -82,9 +82,19 @@ function Runner:_run_git_job()
     pcall(uv.kill, pid)
   end
 
+  local function clean_log(opts_args)
```
Comment (alex-courtis) [inline lua/nvim-tree/git/runner.lua:85]: To clarify: are we just removing nils here?

We could just use a `vim.tbl_filter` if that's the objective.

File: lua/nvim-tree/git/utils.lua [hunk 2]
```diff
@@ -41,7 +41,7 @@ function M.should_show_untracked(cwd)
     return untracked[cwd]
   end
 
-  local cmd = "git -C " .. cwd .. " config --type=bool status.showUntrackedFiles"
+  local cmd = "git -C " .. cwd .. " config status.showUntrackedFiles"
```
Comment (alex-courtis) [inline lua/nvim-tree/git/utils.lua:44]: Thanks for finding this. It looks like I only tested "no" and nil which are the only values that map to a bool.

**Review by alex-courtis — CHANGES_REQUESTED** [review]
- [ ] stylua fixes
- [ ] simplify clean_log

File: lua/nvim-tree/git/runner.lua (hunk 1 above)
Comment (AxLauri) [reply lua/nvim-tree/git/runner.lua]: Yes, just removing the nils, and the suggestion is cleaner. I modified the function to use `vim.tbl_filter`.

**Comment by AxLauri** [comment]
- stylua check should be good now
- simplified clean_log to use `tbl_filter` as suggested

File: lua/nvim-tree/git/runner.lua [hunk 3]
```diff
@@ -82,9 +82,16 @@ function Runner:_run_git_job()
     pcall(uv.kill, pid)
   end
 
+  local function clean_log(t)
+    clean_tbl = vim.tbl_filter(function(v)
```
Comment (alex-courtis) [inline lua/nvim-tree/git/runner.lua:86]: This is a [stylua violation](https://github.com/kyazdani42/nvim-tree.lua/runs/7830009996).

You can simply return the result of tbl_filter.

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Apologies for asking after the first review, but can you please move clean_log to `lua/nvim-tree/utils.lua`

`function M.array_remove_nils(array)`

File: lua/nvim-tree/git/runner.lua (hunk 3 above)
Comment (AxLauri) [reply lua/nvim-tree/git/runner.lua]: - function moved to `M.array_remove_nils(array)` within _lua/nvim-tree/utils.lua`
- stylua & luacheck should be good (I'm on Windows and am having issues with the hooks, so running manually)

**Comment by AxLauri** [comment]
1st PR and new to Lua, so thanks for bearing with me.

**Review by alex-courtis — APPROVED** [review]

**Comment by alex-courtis** [comment]
> 1st PR and new to Lua, so thanks for bearing with me.

It is a pleasure to work with you.

File: lua/nvim-tree/git/runner.lua (hunk 3 above)
Comment (alex-courtis) [reply lua/nvim-tree/git/runner.lua]: > I'm on Windows and am having issues with the hooks, so running manually

If you can fix the hooks I would be grateful; no nvim-tree developers have access or expertise with windows.

**Pull request merged**